src/components/paymentform.main.jsx

```diff
@@ -67,158 +83,246 @@ class PaymentFormMain extends React.Component {
     this.setState({ securityCode: event.target.value });
   }
 
-  submitPayment() {
-    this.cancel();
+  setSaveToProfile(event) {
+    this.setState({ saveToProfile: event.target.checked });
+  }
+
+  submitPayment(event) {
+    event.preventDefault();
+    const {
+      cardHolderName, cardType, cardNumber, securityCode, saveToProfile, paymentForm, orderPaymentForm,
+    } = this.state;
+    if (!cardHolderName || !cardNumber || !securityCode) {
+      this.setState({ failedSubmit: true });
+      return;
+    }
+    let link;
+    if (saveToProfile) {
+      link = paymentForm;
+    } else {
+      link = orderPaymentForm;
+    }
+    let card;
+    switch (cardType) {
+      case 'visa':
+        card = 'Visa';
+        break;
+      case 'master':
+        card = 'MasterCard';
+        break;
+      default:
+        card = 'American Express';
+    }
+    // set link based on savetoprofile
+    login().then(() => {
+      fetch(link, {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+        },
+        body: JSON.stringify({
+          'display-name': `${cardHolderName}'s ${card} ending in: ****${cardNumber.substring(cardNumber.length - 4)}`,
+          token: Math.random().toString(36).substr(2, 9),
```

Comment:
the token you're using is randomly generated for our UI purpose? If so, please document it

Reply:
fixed.